contracts/exchangeIssuance/ExchangeIssuanceZeroEx.sol

```diff
@@ -0,0 +1,561 @@
+        address _weth,
+        IController _setController,
+        address[] memory _issuanceModuleAddresses,
+        bool[] memory _issuanceModuleDebtIssuanceFlags,
+        address _swapTarget
+    )
+        public
+    {
+        require(_issuanceModuleAddresses.length == _issuanceModuleDebtIssuanceFlags.length, "ExchangeIssuance: ISSUANCE MODULE ADDRESSES / TYPE FLAGS LENGTH MISMATCH");
+        setController = _setController;
+
+        WETH = _weth;
+        swapTarget = _swapTarget;
+
+        for (uint256 i = 0; i < _issuanceModuleAddresses.length; i++) {
+            _addIssuanceModule(_issuanceModuleAddresses[i], _issuanceModuleDebtIssuanceFlags[i]);
+        }
+
+    }
+
+    /* ============ External Functions ============ */
+
+    /**
+     * Withdraw slippage to selected address
+     *
+     * @param _tokens    Addresses of tokens to withdraw, specifiy ETH_ADDRESS to withdraw ETH
+     * @param _to        Address to send the tokens to
+     */
+    function withdrawTokens(IERC20[] calldata _tokens, address payable _to) external onlyOwner payable {
+        for(uint256 i = 0; i < _tokens.length; i++) {
+            if(address(_tokens[i]) == ETH_ADDRESS){
+                _to.sendValue(address(this).balance);
+            }
+            else{
+                _tokens[i].safeTransfer(_to, _tokens[i].balanceOf(address(this)));
+            }
+        }
+    }
+
+    receive() external payable {
+        // required for weth.withdraw() to work properly
+        require(msg.sender == WETH, "ExchangeIssuance: Direct deposits not allowed");
+    }
+
+    /* ============ Public Functions ============ */
+
+    /**
+     * Whitelists an issuance module
+     *
+     * @param _issuanceModule    Struct containing data on issuance module to add
+     */
+    function addIssuanceModule(address _issuanceModule, bool _isDebtIssuanceModule) public onlyOwner {
+        _addIssuanceModule(_issuanceModule, _isDebtIssuanceModule);
+    }
+
+    /**
+     * Removes an issuance module from the whitelist
+     *
+     * @param _issuanceModuleAddress    Address of issuance module to remove
+     */
+    function removeIssuanceModule(address _issuanceModuleAddress) public onlyOwner {
+        _removeIssuanceModule(_issuanceModuleAddress);
+    }
+
+    /**
+     * Change the _swapTarget
+     *
+     * @param _swapTarget    Address of the swap target contract. (Usually ZeroEx ExchangeProxy)
+     */
+    function setSwapTarget(address _swapTarget) public onlyOwner {
+        swapTarget = _swapTarget;
+    }
+
+    /**
+     * Runs all the necessary approval functions required for a given ERC20 token.
+     * This function can be called when a new token is added to a SetToken during a
+     * rebalance.
+     *
+     * @param _token    Address of the token which needs approval
+     */
+    function approveToken(IERC20 _token, address _spender) public  isWhitelistedIssuanceModule(_spender) {
+        _safeApprove(_token, _spender, type(uint96).max);
```

Comment:
I see this `uint96` max value in the V1 version of the contract as well. Why is this being used (as opposed to MAX_UINT256?

Reply:
Good question. tbh I kind of "blindly" copied this, but I vaguely remember wondering about this as well.
@ncitron Do you have any insight on this ?
If not I will just change it to max uint256 I guess.

Reply:
Originally it was done since some tokens (like COMP) have a maximum amount of MAX_UINT_96. However, I never liked this since even in the COMP implementation, if you pass in MAX_UINT_256, it would still work. I'm ok with changing it.

Reply:
I just went ahead and changed it to `type(uint256).max`.|